### PR TITLE
feat: Update gRPC service with getUiData call

### PR DESCRIPTION
**Pull request by qiaoweiz**
Related issue: #305 

Files (apis/v0/jvs_plugin_service.pb.go and apis/v0/jvs_plugin_service_grpc.pb.go) are regenerated by command: ```~/projects/jvs/apis/v0$ protoc -I../../protos/v0 --go_out=. --go-grpc_out=. --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative jvs_plugin_service.proto```

### CODE REVIEW

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [inline protos/v0/jvs_plugin_service.proto:53]: Q: Should I use GetUiDataResponse or UiData here? 
background: the guidance from AIP (https://google.aip.dev/131) that UiData can be used directly.

Comment (dandans-verily) [reply protos/v0/jvs_plugin_service.proto:53]: I defer to @yolocs on this question. I personally will use UiData to simplify the logic

File: protos/v0/jvs_plugin_service.proto
Comment (dandans-verily) [inline protos/v0/jvs_plugin_service.proto:28]: Sanity check, is it backward compatible when providing a new rpc without default implementation? My guess is this will break the existing jira plugin since the existing code no longer implement this new interface. If that is the case, may want to research some way to make backward compatible changes

**Review by yolocs — CHANGES_REQUESTED** [review]
You should be able to generate the proto with `go generate ./...` because we have https://github.com/abcxyz/jvs/blob/main/apis/v0/doc.go#L15

File: protos/v0/jvs_plugin_service.proto
Comment (dandans-verily) [reply protos/v0/jvs_plugin_service.proto:49]: `Get` is considered a standard RPC method and cannot be dropped, I would suggest keep `Get`  to avoid confusion. See https://google.aip.dev/131 for more details

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [reply protos/v0/jvs_plugin_service.proto:53]: Update the response name to `UIData`

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [reply protos/v0/jvs_plugin_service.proto:28]: I will update the [plug ](https://github.com/abcxyz/jvs-plugin-jira) shortly after merging this change.

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [reply protos/v0/jvs_plugin_service.proto:49]: Thank you for your suggestions. I agree with @dandans-verily and update the request name to `GetUIDataRequest`.

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [reply protos/v0/jvs_plugin_service.proto:55]: added comments.

File: protos/v0/jvs_plugin_service.proto
Comment (dandans-verily) [reply protos/v0/jvs_plugin_service.proto:28]: Note that you will need to update both of the following places
- jvs-plugin-jira repo
- https://github.com/abcxyz/jvs/blob/main/apis/v0/plugin.go

File: protos/v0/jvs_plugin_service.proto
Comment (dandans-verily) [reply protos/v0/jvs_plugin_service.proto:49]: Any chance you forgot to include the proto changes in the latest commit? I still see GetUiDataResponse

File: protos/v0/jvs_plugin_service.proto
Comment (yolocs) [reply protos/v0/jvs_plugin_service.proto]: Oh I see your point. Maybe we should update https://github.com/abcxyz/jvs/blob/main/apis/v0/plugin.go in this PR (or right after this one)

File: protos/v0/jvs_plugin_service.proto
Comment (yolocs) [reply protos/v0/jvs_plugin_service.proto]: Curious - which part of https://google.aip.dev/131 says we have to prefix it with `Get`?

File: protos/v0/jvs_plugin_service.proto
Comment (yolocs) [inline protos/v0/jvs_plugin_service.proto:58]:
```suggestion
  // The hint for what value to put as the justification.
```

File: protos/v0/jvs_plugin_service.proto
Comment (yolocs) [inline protos/v0/jvs_plugin_service.proto:55]:
```suggestion
  // The display name for the plugin, e.g. for the web UI.
```

File: protos/v0/jvs_plugin_service.proto
Comment (yolocs) [inline protos/v0/jvs_plugin_service.proto:48]:
```suggestion
// GetUIDataRequest is the request to get the plugin data for display purposes.
```

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [reply protos/v0/jvs_plugin_service.proto]: Thank you. I added the empty line.

File: apis/v0/jvs_plugin_service_grpc.pb.go
Comment (dandans-verily) [inline apis/v0/jvs_plugin_service_grpc.pb.go:1]: You actually will need to keep all the Copyright stuff since it is how open source works

File: apis/v0/jvs_service_grpc.pb.go
Comment (dandans-verily) [inline apis/v0/jvs_service_grpc.pb.go:1]: Not sure whether you want to change other generated files. I will let @yolocs comment on this part

**Review by dandans-verily — APPROVED** [review]
LGTM with two nits

File: apis/v0/jvs_service_grpc.pb.go
Comment (yolocs) [reply apis/v0/jvs_service_grpc.pb.go:1]: Fine with that

**Review by yolocs — APPROVED** [review]

File: apis/v0/jvs_plugin_service_grpc.pb.go
Comment (yolocs) [reply apis/v0/jvs_plugin_service_grpc.pb.go:1]: Oh yeah, good point!

File: apis/v0/jvs_service_grpc.pb.go
Comment (qiaoweiz) [reply apis/v0/jvs_service_grpc.pb.go:1]: Thank you! I will keep other generated files in this pr. (Marked this conversation as resolved.

File: apis/v0/jvs_plugin_service_grpc.pb.go
Comment (qiaoweiz) [reply apis/v0/jvs_plugin_service_grpc.pb.go:1]: I'm unsure about the reason behind the removal of the Copyright information by the new command.

I noticed that the apis/v0/cert_action_service_grpc.pb.go and apis/v0/jvs_service_grpc.pb.go files did not originally contain the "certificate" component. Could you please confirm if this aligns with the expected behavior?

Comment (yolocs) [reply apis/v0/jvs_plugin_service_grpc.pb.go:1]: Good catch. They all should have the copyright header.

File: protos/v0/jvs_plugin_service.proto
Comment (qiaoweiz) [reply protos/v0/jvs_plugin_service.proto]: Thank you! I will update https://github.com/abcxyz/jvs/blob/main/apis/v0/plugin.go in the next one.

File: apis/v0/jvs_plugin_service_grpc.pb.go
Comment (qiaoweiz) [reply apis/v0/jvs_plugin_service_grpc.pb.go:1]: Added the copyright. 

And updated the year of copyright in some files from `2022` to `2023`.

**Review by dandans-verily — APPROVED** [review]

File: apis/v0/jwt.go
Comment (yolocs) [inline apis/v0/jwt.go:1]: btw it's not required to keep the year up to date unless you're adding a new file

**Review by yolocs — APPROVED** [review]

File: apis/v0/jwt.go
Comment (qiaoweiz) [reply apis/v0/jwt.go:1]: Got it. Thank you!

**Pull request merged**
